driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryption25LookupProseTests.java

```diff
@@ -0,0 +1,251 @@
+
+        encryptedDb.createCollection("qe",
+                new CreateCollectionOptions().encryptedFields(bsonDocumentFromPath("schema-qe.json")));
+        encryptedDb.createCollection("qe2",
+                new CreateCollectionOptions().encryptedFields(bsonDocumentFromPath("schema-qe2.json")));
+
+        encryptedDb.createCollection("no_schema");
+        encryptedDb.createCollection("no_schema2");
+
+        // Insert documents with `encryptedClient`:
+        Consumer<String> insert = (name) -> {
+            encryptedDb.getCollection(name).insertOne(new Document(name, name));
+        };
+        insert.accept("csfle");
+        insert.accept("csfle2");
+        insert.accept("qe");
+        insert.accept("qe2");
+        insert.accept("no_schema");
+        insert.accept("no_schema2");
+
+        // Create an unencrypted MongoClient named `unencryptedClient`.
+        MongoClient unencryptedClient = createMongoClient(getMongoClientSettingsBuilder().build());
+        MongoDatabase unencryptedDb = unencryptedClient.getDatabase("db");
+
+        Consumer<String> assertDocument = (name) -> {
+            List<BsonDocument> pipeline = Arrays.asList(
+                    BsonDocument.parse("{\"$project\" : {\"_id\" : 0, \"__safeContent__\" : 0}}")
+            );
+            Document decryptedDoc = encryptedDb.getCollection(name)
+                    .aggregate(pipeline).first();
+            assertEquals(decryptedDoc, new Document(name, name));
+            Document encryptedDoc = unencryptedDb.getCollection(name)
+                    .aggregate(pipeline).first();
+            assertNotNull(encryptedDoc);
+            assertEquals(Binary.class, encryptedDoc.get(name).getClass());
+        };
+
+        assertDocument.accept("csfle");
+        assertDocument.accept("csfle2");
+        assertDocument.accept("qe");
+        assertDocument.accept("qe2");
+
+        unencryptedClient.close();
+        encryptedClient.close();
+
+        client = createMongoClient(getMongoClientSettingsBuilder()
+                .autoEncryptionSettings(
+                        AutoEncryptionSettings.builder()
+                                .keyVaultNamespace(dataKeysNamespace.getFullName())
+                                .kmsProviders(kmsProviders)
+                                .build())
+                .build());
+    }
+
+    @AfterEach
+    @SuppressWarnings("try")
+    public void cleanUp() {
+        //noinspection EmptyTryBlock
+        try (MongoClient ignored = this.client) {
+            // just using try-with-resources to ensure they all get closed, even in the case of exceptions
+        }
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "csfle, no_schema",
+            "qe, no_schema",
+            "no_schema, csfle",
+            "no_schema, qe",
+            "csfle, csfle2",
+            "qe, qe2",
+            "no_schema, no_schema2"})
+    void cases1Through7(final String from, final String to) {
```

Comment:
[nit] Just a bit more context that the test is about using multiple auto-encryption schemas. It might save time understanding its purpose in the future.
```suggestion
    void shouldUseMultipleAutoEncryptionSchemasCases1Through7(final String from, final String to) {
```

Reply:
I used the "test" prefix.

For prose tests, I think the priority is being able to easily reference the prose test spec file. Tests should be numbered, with the number near the start of the test or file to ensure alphabetical sorting. If extra wording will help clarify, tests should match spec wording of the individual test (so tests should end up with mostly different names), and if the spec's test case naming is unclear, an edit should be made to the spec.